lib/elixir/lib/file/stream.ex

```diff
@@ -73,7 +73,8 @@ defmodule File.Stream do
       start_fun =
         fn ->
           case :file.open(path, read_modes(modes)) do
-            {:ok, device}    -> device
+            {:ok, device}    ->
+              if :strip_bom in modes, do: strip_bom(device), else: device
```

Comment:
The current code is fine. Unfortunately we cannot write the proposed approach because that's not an valid guard. :(